server/src/main/java/org/elasticsearch/cluster/metadata/MetadataCreateIndexService.java

```diff
@@ -517,7 +517,8 @@ private ClusterState applyCreateIndexWithTemporaryService(
                     temporaryIndexMeta.getRoutingNumShards(),
                     sourceMetadata,
                     temporaryIndexMeta.isSystem(),
-                    currentState.getMinTransportVersion()
+                    currentState.getMinTransportVersion(),
+                    request.dataStreamName() != null
```

Comment:
I see in other places in `MetadataCreateIndexService` we check if `dataStreamName` is not `null` to check if   we are dealing with a data stream or not.